lib/lobby.ts

```diff
@@ -0,0 +1,172 @@
+import { getClient } from '.';
+import { DbItem, IDbItem } from './db-item';
+import { COLLECTION } from './private/enums';
+import { User } from './user';
+
+type DatabaseEntry = Omit<ILobby, 'collectionName'>;
+type ClientResponse = Omit<DatabaseEntry, 'users' | 'uri'>;
+
+export interface LobbyProps {
+  /**
+  * The spotify id of playlist
+  */
+  readonly spotifyPlaylistId?: string;
+
+  /**
+   * The manager of a lobby
+   */
+  readonly managerId: string;
+
+  /**
+   * The participants in a lobby
+   */
+  readonly participants: string[];
+
+  /**
+   * The theme of a lobby
+   */
+  readonly theme: string;
+
+  /**
+   * The playlist name
+   */
+  readonly name: string;
+
+  /**
+   * The list of song ids in the playlist
+   */
+  readonly songIds: string[];
+}
+
+export interface ILobby extends LobbyProps, IDbItem {
+}
+
+/**
+ * The class containing a user and their data
+ */
+export class Lobby extends DbItem implements ILobby {
+  /**
+   * A static function to query for a user from their id
+   *
+   * @returns a user object if the id exists in the database
+   */
+  public static async fromId(id: string): Promise<Lobby | null> {
+    const client = await getClient();
+    const document = await client.findDbItem(COLLECTION.LOBBIES, id);
+    if (!document) return null;
+    const content: DatabaseEntry = document.getContent() as DatabaseEntry;
+    return new Lobby(id, content, document.key ?? null);
+  }
+
+  /**
+   * The spotify id of playlist
+   */
+  readonly spotifyPlaylistId?: string;
+
+  /**
+   * The manager user id of a lobby
+   */
+  readonly managerId: string;
+
+  /**
+   * The theme of a lobby
+   */
+  readonly theme: string;
+
+  /**
+   * The playlist name
+   */
+  readonly name: string;
+
+  /**
+   * The participants in a lobby
+   */
+  get participants(): string[] {
+    return this.#participants;
+  }
+
+  #participants: string[];
+
+  /**
+   * The list of song ids in the playlist
+   */
+  get songIds(): string[] {
+    return this.#songIds;
+  }
```

Comment:
why is this set up like this? why cant we do a public readonly of just the values?

Reply:
Oh I think u did this for topSongs in User

Reply:
nah ur right this is good!